engine/src/main/java/com/redhat/cloud/notifications/models/ApplicationEmailTemplateId.java

```diff
@@ -0,0 +1,41 @@
+package com.redhat.cloud.notifications.models;
+
+import com.redhat.cloud.notifications.db.converters.EmailSubscriptionTypeConverter;
+
+import javax.persistence.Convert;
+import javax.persistence.Embeddable;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.UUID;
+
+@Embeddable
+public class ApplicationEmailTemplateId implements Serializable {
+
+    @NotNull
+    public UUID applicationId;
+
+    @NotNull
+    @Size(max = 50)
+    @Convert(converter = EmailSubscriptionTypeConverter.class)
+    public EmailSubscriptionType subscriptionType;
```

Comment:
Here's how I imagine things regarding the instant notification templates at application/bundle level:
- We would first look for an "instant" template at event type level in the DB.
- If not found, we would look for an "instant" template at application level.
- If not found, same thing at bundle level in the DB.

Reply:
Templates are not changing a lot, so we would cache the templates for 1 day to limit the number of SQL queries and clear the cache on template update.

Reply:
Better naming proposal:
- EventTypeInstantEmailTemplate (currently `EventTypeEmailTemplate`)
- ApplicationInstantEmailTemplate (fallback if nothing found at event type level, not created yet)
- BundleInstantEmailTemplate (fallback if nothing found at app level, not created yet)
- ApplicationDailyEmailTemplate (currenlty `ApplicationEmailTemplate`)
- BundleDailyEmailTemplate (fallback if nothing found at app level, not created yet)

Reply:
@pilhuhn You may have suggestions/ideas about this.

Reply:
I'm not really sure if we want to add the fallback rules. 
I don't think we would like to fallback to bundle globals or even app globals templates. 

Right now the templates are tighly coupled with the payload, but nothing is enforcing that the data is being consumed correctly (inside the template). Qute provides a way to validate - but we would need to have a schema in place first.

I think is best if the apps select a specific template for each event (it could indeed be the same template, but they would select and test it manually).

Besides, a global template won't be user friendly or contain pertinent information IMO - I think it would be more useful when onboarding, and we can do something simpler like `if !templateFound && env != prod => use_onboarding_template`.

About new types, we might add a weekly subscription type and we would indeed support it only at the application level.
We can keep the `subscriptionType` param to keep track of this. 

Would like to hear other's thoughts about this.

Reply:
If fallbacks were implemented, there would certainly be a condition (boolean field) to enable/disable them depending on the context, and they would probably be disabled most of the time.

But maybe we don't need fallbacks at all. It's not in that PR anyway 😄